Build failure: pkgsCross.mingw{32,W64}.gettext

### Describe the bug
mingw32 build support for gettext is broken. There are conditionals within the gettext derivation that leads me to believe that mingw32 builds should be working.

### Steps To Reproduce
Steps to reproduce the behavior:
1. `nix build 'nixpkgs#pkgs.pkgsCross.mingw32.gettext'`

### Expected behavior
The package to be built with not error

### Screenshots
If applicable, add screenshots to help explain your problem.

### Additional context
Here is the error that I get 
```
error: builder for '/nix/store/4y9nvkp5pq5p8p7238hwasjkk1vni9i1-gettext-i686-w64-mingw32-0.21.1.drv' failed with exit code 1;
       last 10 log lines:
       > patching file gettext-runtime/src/gettext.sh.in
       > Hunk #1 succeeded at 85 with fuzz 1 (offset -1 lines).
       > applying patch /nix/store/yqwx9yln5i68nw61mmp9gz066yz3ri99-0001-msginit-Do-not-use-POT-Creation-Date.patch
       > patching file gettext-tools/src/msginit.c
       > applying patch /nix/store/8g74yjpdnxbvc855rb5y8q8j7y5bd04d-gettext_formatstring-ruby.patch
       > patching file gettext-tools/woe32dll/gettextsrc-exports.c
       > Reversed (or previously applied) patch detected!  Assume -R? [n]
       > Apply anyway? [n]
       > Skipping patch.
       > 1 out of 1 hunk ignored -- saving rejects to file gettext-tools/woe32dll/gettextsrc-exports.c.rej
       For full logs, run 'nix log /nix/store/4y9nvkp5pq5p8p7238hwasjkk1vni9i1-gettext-i686-w64-mingw32-0.21.1.drv'.
```

Looks like the changes for one of the patches was upstreamed. If I remove the patch (this is okay because it is now in the source code), I will get the following error.
```
/nix/store/f6y9zp2yj7bm44gx3h951w5ad1dnn1qc-i686-w64-mingw32-binutils-2.40/bin/i686-w64-mingw32-ld: .libs/libgettextsrc_la-write-catalog.o: in function `msgdomain_list_print':
/build/gettext-0.21.1/gettext-tools/src/write-catalog.c:251: undefined reference to `close_used_without_requesting_gnulib_module_close'
```

### Notify maintainers
@Artturin 
<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
output here
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/libraries/gettext/default.nix
+++ b/pkgs/development/libraries/gettext/default.nix
@@ -1,5 +1,4 @@
-{ stdenv, lib, fetchurl, fetchpatch, libiconv, xz, bash
-, gnulib
+{ stdenv, lib, fetchurl, libiconv, xz, bash
 }:
 
 # Note: this package is used for bootstrapping fetchurl, and thus
@@ -20,11 +19,7 @@ stdenv.mkDerivation rec {
     # fix reproducibile output, in particular in the grub2 build
     # https://savannah.gnu.org/bugs/index.php?59658
     ./0001-msginit-Do-not-use-POT-Creation-Date.patch
-  ] ++ lib.optional stdenv.hostPlatform.isWindows (fetchpatch {
-    url = "https://aur.archlinux.org/cgit/aur.git/plain/gettext_formatstring-ruby.patch?h=mingw-w64-gettext&id=e8b577ee3d399518d005e33613f23363a7df07ee";
-    name = "gettext_formatstring-ruby.patch";
-    sha256 = "sha256-6SxZObOMkQDxuKJuJY+mQ/VuJJxSeGbf97J8ZZddCV0=";
-  });
+  ];
 
   outputs = [ "out" "man" "doc" "info" ];
 
@@ -50,6 +45,8 @@ stdenv.mkDerivation rec {
   '' + lib.optionalString stdenv.hostPlatform.isCygwin ''
     sed -i -e "s/\(cldr_plurals_LDADD = \)/\\1..\/gnulib-lib\/libxml_rpl.la /" gettext-tools/src/Makefile.in
     sed -i -e "s/\(libgettextsrc_la_LDFLAGS = \)/\\1..\/gnulib-lib\/libxml_rpl.la /" gettext-tools/src/Makefile.in
+  '' + lib.optionalString stdenv.hostPlatform.isMinGW ''
+    sed -i "s/@GNULIB_CLOSE@/1/" */*/unistd.in.h
   '';
 
   strictDeps = true;

